lib/ruby_indexer/lib/ruby_indexer/reference_finder.rb

```diff
@@ -321,7 +325,10 @@ def collect_constant_references(name, location)
     def collect_instance_variable_references(name, location, declaration)
       return unless @target.is_a?(InstanceVariableTarget) && name == @target.name
 
-      @references << Reference.new(name, location, declaration: declaration)
+      receiver_type = @stack.join("::")
```

Comment:
This helper method handles the namespaces considering top level references too.

```ruby
module Foo
  class ::Bar; end
end
```

```suggestion
      receiver_type = Index.actual_nesting(@stack, nil)
```

Reply:
Thanks!